### ISSUE TITLE
BadIdentityTokenRejected when trying to connect to opcua server using login/password.

### ISSUE THREAD

**Issue by xydan83** [open62541/open62541#6823]
## Checklist

Please provide the following information:

 - [x] open62541 Version (release number or git tag): v1.4.6
 - [x] Other OPC UA SDKs used (client or server): client
 - [x] Operating system: Ubuntu 22.04.5

Hi!
I encountered some problem when trying to connect to the server using login/password.
I chose UAServer (1.8.0) as a test server.

When trying to connect via UAExpert (v1.7.1 540) everything is fine:
![image](https://github.com/user-attachments/assets/88a0dfa7-ea19-4741-b58e-a8849b38f094)

But if I try to connect using open62541 example (client_connect), I see error: BadIdentityTokenRejected
![image](https://github.com/user-attachments/assets/99038520-b7de-47e5-a6c8-f591a17778eb)

As you can see, the library can't find the 0 UserTokenPolicy endpoint.

Is this some possible compatibility issue or some other problem?

**Comment by lesslefou** [comment]
I encountere the same problem... any news ?

**Comment by xydan83** [comment]
> I encounter the same problem... any news ?

Yes. Some OPC UA servers require encryption of not only authentication certificates, but also password data. For this, you need to use certificates. Use **UA_ClientConfig_setDefaultEncryption** for this. 
Then you can use **UA_CertificateVerification_AcceptAll** if you accepted all servers certificate.
Don`t forget about applicationUri.
And then **UA_ClientConfig_setAuthenticationUsername**.

**Comment by ffreality** [comment]
> > I encounter the same problem... any news ?
> 
> Yes. Some OPC UA servers require encryption of not only authentication certificates, but also password data. For this, you need to use certificates. Use **UA_ClientConfig_setDefaultEncryption** for this. Then you can use **UA_CertificateVerification_AcceptAll** if you accepted all servers certificate. Don`t forget about applicationUri. And then **UA_ClientConfig_setAuthenticationUsername**.

`UA_CertificateVerification_AcceptAll` is for older versions. It is deprecated right now. I defined ``applicationUri`` based on opc ua v1.4.0 rules and wrote verification logic like this but I receive `BadIdentityTokenRejected`.

```
UA_ClientConfig_setAuthenticationUsername(ClientConfig, UserName, Password);
UA_ClientConfig_setAuthenticationCert(ClientConfig, Certificate, PrivateKey);
UA_ClientConfig_setDefaultEncryption(ClientConfig, Certificate, PrivateKey, List_Trust, In_Params.TrustListSize, List_Revoke, In_Params.RevokeListSize);
UA_CertificateGroup_AcceptAll(&ClientConfig->certificateVerification);
```

Do you know the solution ?

**Comment by xydan83** [comment]
> > > I encounter the same problem... any news ?
> > 
> > 
> > Yes. Some OPC UA servers require encryption of not only authentication certificates, but also password data. For this, you need to use certificates. Use **UA_ClientConfig_setDefaultEncryption** for this. Then you can use **UA_CertificateVerification_AcceptAll** if you accepted all servers certificate. Don`t forget about applicationUri. And then **UA_ClientConfig_setAuthenticationUsername**.
> 
> `UA_CertificateVerification_AcceptAll` is for older versions. It is deprecated right now. I defined `applicationUri` based on opc ua v1.4.0 rules and wrote verification logic like this but I receive `BadIdentityTokenRejected`.
> 
> ```
> UA_ClientConfig_setAuthenticationUsername(ClientConfig, UserName, Password);
> UA_ClientConfig_setAuthenticationCert(ClientConfig, Certificate, PrivateKey);
> UA_ClientConfig_setDefaultEncryption(ClientConfig, Certificate, PrivateKey, List_Trust, In_Params.TrustListSize, List_Revoke, In_Params.RevokeListSize);
> UA_CertificateGroup_AcceptAll(&ClientConfig->certificateVerification);
> ```
> 
> Do you know the solution ?

Hi! 

I answered you in your new issue: https://github.com/open62541/open62541/issues/7702
